src/torrent/download/resource_manager.cc

```diff
@@ -343,41 +339,33 @@ ResourceManager::balance_unchoked(unsigned int weight, unsigned int max_unchoked
   // that won't work as they need to choke peers once their priority
   // is turned off.
 
-  choke_group* choke_groups[group_size()];
-  std::copy(choke_base_type::begin(), choke_base_type::end(), choke_groups);
+  auto choke_groups = std::vector<choke_group*>(group_size());
+  std::copy(choke_base_type::begin(), choke_base_type::end(), std::back_inserter(choke_groups));
```

Comment:
We know the size, so this should preallocate choke_groups and do straight copy without back_inserter.

Reply:
I think this is what you meant.